python-sdk/src/astro/sql/operators/raw_sql.py

```diff
@@ -69,6 +96,52 @@ def make_row_serializable(rows: Any) -> Any:
             return [SdkLegacyRow.from_legacy_row(r) if isinstance(r, SQLAlcRow) else r for r in rows]
         return rows
 
+    @staticmethod
+    def results_as_list(result: ResultProxy) -> list:
+        """
+        Convert the result of a SQL query to a list
+        """
+        return cast(List[Any], result.fetchall())
+
+    @staticmethod
+    def results_as_pandas_dataframe(result: ResultProxy) -> pd.DataFrame:
+        """
+        Convert the result of a SQL query to a pandas dataframe
+        """
+        return pd.DataFrame(result.fetchall(), columns=result.keys())
+
+    def get_results_format_handler(self, results_format: str):
+        return getattr(self, f"results_as_{results_format}")
+
+    def get_handler(self):
+        # if a user specifies a results_format, we will return the results in that format
+        # note that it takes precedence over the handler
+        if self.results_format:
+            self.handler = self.get_results_format_handler(results_format=self.results_format)
+        if self.handler:
+            self.handler = self.get_wrapped_handler(
+                fail_on_empty=self.fail_on_empty, conversion_func=self.handler
+            )
+        return self.handler
+
+    @staticmethod
+    def get_wrapped_handler(fail_on_empty: bool, conversion_func: Callable):
+        # if fail_on_empty is set to False, wrap in a try/except block
+        # this is because the conversion function will fail if the result is empty
+        # due to sqlalchemy failing when there are no results
+        if not fail_on_empty:
+
+            def handle_exceptions(result):
+                try:
+                    return conversion_func(result)
+                except Exception:  # skipcq: PYL-W0703
+                    return None
+
+            return handle_exceptions
```

Comment:
We should atleast add some logging when it fails

Reply:
@kaxil Maybe we can add a log saying that the exception is handled because of `fail_on_empty`?

Reply:
@kaxil I'm going ahead with the above suggestions, if you feel this is not right, I'll raise a separate PR for this.